### ISSUE TITLE
IOS Problems Help me

### ISSUE THREAD

**Issue by mertbuldur** [dooboolab/react-native-iap#791]
Hi  I have some problems for ios .
With android   **const data = await RNIap.consumePurchaseAndroid(x.purchaseToken);** method with the return of the data to the database to save my information and I can save. How can I do this in IOS?
What is RNIap.consumeAllItemsAndroid(); in ios?

Thanks for help
Version : 4.0.3

**Comment by hyochan** [comment]
Related to #783 I think. You can use `finishTransactionIOS` or `finishTransaction` together.

**Comment by mertbuldur** [comment]
My Code is 
`componentDidMount = async () => {
        try {
            const result = await RNIap.initConnection();
            await RNIap.consumeAllItemsAndroid();
                  RNIap.clearProductsIOS();




        } catch (err) {
            //Alert.alert("Hata 1!",err.message);
        }

        try
        {
            const packages = await RNIap.getProducts(items);

            this.setState({
                packages,
            })
        }
        catch (e) {
           Alert.alert("Hata 2!",e.message);
        }


        purchaseUpdateSubscription = purchaseUpdatedListener(
            async (purchase: ProductPurchase) => {
                if (purchase.purchaseStateAndroid === 1 && !purchase.isAcknowledgedAndroid) {
                    try {
                            const ackResult = await acknowledgePurchaseAndroid(
                                purchase.purchaseToken,
                            );
                    } catch (ackErr) {
                        console.log('x');
                        console.warn('ackErr', ackErr);
                    }
                }
                this.setState({ receipt: purchase }, () =>
                    this.goNext(),
                );
            },
        );


        purchaseErrorSubscription = purchaseErrorListener(
            (error: PurchaseError) => {
                console.log('y');
                Alert.alert('purchase error', JSON.stringify(error));
            },
        );

    };

goNext = async () => {



        if(Platform.OS == 'android') {
            const x = JSON.parse(this.state.receipt.transactionReceipt);
            const data = await RNIap.consumePurchaseAndroid(x.purchaseToken);
            if (data.code === 'OK') {
                axios.post(`${API_URL}/api/user/payment`, {
                    access_token: this.props.AuthStore.token,
                    packageId: this.state.selectedPackageName
                }).then(res => {

                    let data = JSON.parse(this.props.UserStore.info);
                    data['premium'] = true;
                    this.props.UserStore.saveInfo(data);
                    this.props.navigation.navigate('Profile');
                });
            } else {
                Alert.alert('Hata!', 'Bir problem oldu lütfen daha sonra deneyiniz');
            }
        }
        else
        {

             const x = this.state.receipt;
             //const l = await RNIap.getReceiptIOS();
            const data = await RNIap.finishTransactionIOS(x.transactionId);

        }

    };
 `

Is it alright ?

**Comment by TiagoGouvea** [comment]
I'm facing something (maybe) related to this, that I cannot understand already.

I subscribed on a product on my sandbox, on iOS. All worked like expected. I didn't finish that transaction.

After that, on every reload, I called `getAvailablePurchases`  and received a new *one* purchase as result. Like, 6 reloads, 6 purchases. Each one with different `transactionId`, but same `originalTransactionIdentifierIOS`. 

I added the `finishTransactionIOS` call on my third reload.

I don't know exactly why it stopped to return new purchases. But, I'm still receiving 6 purchases on my `getAvailablePurchases`.

I tried to call `finishTransactionIOS` on each purchase.transactionId, but I still receiving back that 6 purchases.

What's wrong? Did I miss something?

Thanks for any help
